src/test/scala/chiselTests/LFSR16.scala

```diff
@@ -6,43 +6,45 @@ import chisel3._
 import chisel3.testers.BasicTester
 import chisel3.util._
 
-class LFSR16 extends Module {
-  val io = IO(new Bundle {
-    val inc = Input(Bool())
-    val out = Output(UInt(16.W))
-  })
-  val res = RegInit(1.U(16.W))
-  when (io.inc) {
-    val nxt_res = Cat(res(0)^res(2)^res(3)^res(5), res(15,1))
-    res := nxt_res
+/**
+  * This test creates two 4 sided dice.
+  * each cycle it adds them together and adds a count to the bin corresponding to that value
+  * The asserts check that the bins show the correct distribution.
+  */
```

Comment:
A possibly better test to consider would be to check for maximum periodicity. I.e., start it at 1 and see how many cycles it takes to get back to 1. That breaks down for bigger LFSRs, but as this is only 16-bit it's tractable.

Reply:
I can do this if you think it would be better. I kind of like the distribution test as it gives a bit more of the flavor of how one might use the block.

Reply:
Maybe both? I think the example-motivated (craps odds calculating) approach is great and pays dividends long-term. From personal experience, it's super easy to screw up LFSR taps, though.

If you want a tiny test, how about:
```scala
class LFSR16MaxPeriod extends BasicTester {

  val lfsr16 = LFSR16()
  val started = RegNext(true.B, false.B)

  val (_, wrap) = Counter(started, math.pow(2.0, 16).toInt - 1)
  when (lfsr16 === 1.U && started) {
    chisel3.assert(wrap)
    stop()
  }

}
```

Reply:
Added your test too